roles/sugarizer/tasks/main.yml

```diff
@@ -19,25 +19,22 @@
     quiet: yes
 
 
-# 3 stanzas moved up from install.yml, so Debian finishes "BIG" IIAB install:
-# (WITH LOUD RED WARNINGS!)
+# 3 stanzas moved up from install.yml, so Debian-or-any-OS-where-MongoDB-fails
+# still finish their "BIG-sized" IIAB install: (WITH LOUD RED WARNINGS!)
 
-# If we got here sugarizer_install is True
 - name: "Set 'mongodb_install: True'"
   set_fact:
     mongodb_install: True
 
-# MongoDB auto-started by sugarizer.service - let's set 'mongodb_enabled: True'
-# regardless - anticipating that MongoDB (should!) run on demand:
-- name: "Set 'mongodb_enabled: True' when sugarizer_enabled"
-  set_fact:
```

Comment:
This is longer true once you invoke the systemd module with "enabled: yes" in mongodb role.

Reply:
This PR completely avoids enabling MongoDB.

As explained in several Ansible `name:` comments — the idea is to keep Ansible variable `mongodb_enabled` more properly in sync with the output of systemd's `systemctl is-enabled mongodb` — as the prior code left them very confusingly out of sync.

So while I'm not in full agreement that `mongodb.sugarizer` should remain disabled, my goal here was only to make the code more readable — while preserving its current mechanics as much as possible — i.e. respecting de facto status of systemd's `is-enabled` and bringing Ansible car `mongodb_enabled` into line with de facto current practice.

This allows MongoDB to be easily enabled in future (both on the Ansible side and the systemd side, keeping both in sync going forward to maintain sanity) — if that's indeed thought to be preferable.

Reply:
Aside: the reason we might want to enable MongoDB in future (on the Ansible side and systemd side, keeping both in sync to maintain sanity!) is to make MongoDB more reliable.

e.g. for IIAB installations that are not constantly being rebooted, but rather are left on for many weeks like a traditional server.

Allowing Sugarizer (and potentially other apps using MongoDB) to become more reliable.

Reply:
> This PR completely avoids enabling MongoDB.
> 
No, there is code that will, just currently unused.

> As explained in several Ansible `name:` comments — the idea is to keep Ansible variable `mongodb_enabled` more properly in sync with the output of systemd's `systemctl is-enabled mongodb` — as the prior code left them very confusingly out of sync.
> 

Why so? I was assuming the iiab.ini file was recording the "expected state of the service when booted", how the service is started is irrelevant for that purpose.

> So while I'm not in full agreement that `mongodb.service` should remain disabled, my goal here was only to make the code more readable — while preserving its current mechanics as much as possible — i.e. respecting de facto status of systemd's `is-enabled` and bringing Ansible var `mongodb_enabled` into line with de facto current practice.
> 

The pertinent information is "should the service be running when booted" not one of if systemd should automaticlly start the service standalone, as the service is started on demand by sugarizer.

> This allows MongoDB to be easily enabled in future (both on the Ansible side and the systemd side, keeping both in sync going forward to maintain sanity) — if enabling it is indeed thought to be preferable[*].
> 
> [*] e.g. for purposes of MongoDB reliability.

If you think so but whatever.

Reply:
> Aside: the reason we might want to enable MongoDB in future (on the Ansible side and systemd side, keeping both in sync as much as possible to maintain sanity!) is to make MongoDB more reliable.
>
More reliable in which way?
 
> e.g. for IIAB installations that are not constantly being rebooted, but rather are left on for many weeks like a traditional server.
> 
Now I'm confused, once the service starts the service is running, what are you referring to? 

> Allowing Sugarizer (and potentially other apps using MongoDB) to become more reliable.

If you want other apps to easily integrate then maybe look at port usage issue I pointed out.